src/VisualQueryBuilder/components/LabelFilters.tsx

```diff
@@ -0,0 +1,85 @@
+import { isEqual } from 'lodash';
+import React, { useEffect, useState } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+
+import { QueryBuilderLabelFilter } from '../types';
+
+import { EditorField, } from '../../QueryEditor/EditorField';
+import { EditorFieldGroup } from '../../QueryEditor/EditorFieldGroup';
+import { EditorList } from '../../QueryEditor/EditorList';
+import { LabelFilterItem } from './LabelFilterItem';
+
+
+export const MISSING_LABEL_FILTER_ERROR_MESSAGE = 'Select at least 1 label filter (label and value)';
+
+export interface Props {
+  labelsFilters: QueryBuilderLabelFilter[];
+  onChange: (labelFilters: QueryBuilderLabelFilter[]) => void;
+  onGetLabelNames: (forLabel: Partial<QueryBuilderLabelFilter>) => Promise<SelectableValue[]>;
+  onGetLabelValues: (forLabel: Partial<QueryBuilderLabelFilter>) => Promise<SelectableValue[]>;
+  /** If set to true, component will show error message until at least 1 filter is selected */
+  labelFilterRequired?: boolean;
+  multiValueSeparator?: string;
+}
+
+export function LabelFilters({
+  labelsFilters,
+  onChange,
+  onGetLabelNames,
+  onGetLabelValues,
+  labelFilterRequired,
+  multiValueSeparator,
+}: Props) {
+  const defaultOp = '=';
+  const [items, setItems] = useState<Array<Partial<QueryBuilderLabelFilter>>>([{ op: defaultOp }]);
+
+  useEffect(() => {
+    if (labelsFilters.length > 0) {
+      setItems(labelsFilters);
+    } else {
+      setItems([{ op: defaultOp }]);
+    }
+  }, [labelsFilters]);
+
+  const onLabelsChange = (newItems: Array<Partial<QueryBuilderLabelFilter>>) => {
+    setItems(newItems);
+
+    // Extract full label filters with both label & value
+    const newLabels = newItems.filter((x) => x.label != null && x.value != null);
+    if (!isEqual(newLabels, labelsFilters)) {
+      onChange(newLabels as QueryBuilderLabelFilter[]);
```

Comment:
Another type to add to the follow up.

Reply:
Added to https://github.com/grafana/grafana-experimental/issues/101